libs/blocks/marketo/marketo.js

```diff
@@ -59,8 +65,24 @@ export const decorateURL = (destination, baseURL = window.location) => {
   return null;
 };
 
-export const formSuccess = (form) => {
-  const formEl = form.getFormElem().get(0);
+const setPreference = (key, value) => {
+  if (value && key?.includes('.')) {
+    const keyParts = key.split('.');
+    const lastKey = keyParts.pop();
+    const formDataObject = keyParts.reduce((obj, part) => {
+      obj[part] = obj[part] || {};
+      return obj[part];
+    }, window.mcz_marketoForm_pref);
+    formDataObject[lastKey] = value;
+  }
+};
```

Comment:
I prefer early returns rather than if's
```suggestion
const setPreference = (key = "", value) => {
  if(!value || !key.includes('.')) return
  const keyParts = key.split('.');
  const lastKey = keyParts.pop();
  const formDataObject = keyParts.reduce((obj, part) => {
    obj[part] = obj[part] || {};
    return obj[part];
  }, window.mcz_marketoForm_pref);
  formDataObject[lastKey] = value;
};
```

Reply:
Looks good, thank you